python/lsst/daf/butler/core/dimensions/sets.py

```diff
@@ -402,57 +208,153 @@ def findIf(self, predicate, default=None):
             return default
         return t[0]
 
+    def __len__(self):
+        return len(self._elements)
 
-class DimensionNameSet(DimensionSetBase):
-    r"""An incomplete, name-only stand-in for `DimensionSet` or
-    `DimensionGraph`.
+    def __str__(self):
+        return "{{{}}}".format(", ".join(self._elements.keys()))
 
-    Parameters
-    ----------
-    names : iterable of `str`
-        The names of elements to conceptually include in the set.
+    def __hash__(self):
+        return hash(tuple(self._elements.keys()))
```

Comment:
Don't you need `sorted()` somewhere here?

Reply:
Happily, no.  Now that there's no unnormalized `DimensionNameSet` anymore, the ordering of the keys is deterministic (topological and then lexicographically sorted).

Reply:
Maybe add a comment here so that I don't ask the same question again in the next review?